src/libraries/System.IO.Ports/src/System/IO/Ports/SerialPort.cs

```diff
@@ -963,7 +963,7 @@ public string ReadExisting()
                 Buffer.BlockCopy(_inBuffer, _readPos, bytesReceived, 0, CachedBytesToRead);
             }
 
-            _internalSerialStream.Read(bytesReceived, CachedBytesToRead, bytesReceived.Length - (CachedBytesToRead));    // get everything
+            _internalSerialStream.ReadExactly(bytesReceived, CachedBytesToRead, bytesReceived.Length - (CachedBytesToRead));    // get everything
```

Comment:
This won't compile as written. This library builds for multiple target frameworks, some of which don't have ReadExactly.

Reply:
Thanks, I've added a check for `NET7_0_OR_GREATER` before using `ReadExactly`.

Is there a build target to check locally if I've used a function that is not available in a target framework? I ran the tests locally with `.\build.cmd -subset clr+mono+libs+libs.tests -test -rc Release` but got no build error.

Reply:
Thanks. That will only end up building for the current platform. There used to be a command for building all flavors of all libraries, but I can't find it in the docs now. @ViktorHofer? In the meantime, you can cd into the src folder for the library and `dotnet build` will build all targets for it.

Reply:
`build.cmd libs -allconfigurations`